### PR TITLE
#2243 redefine the nodeNames and deadNodes for Stream cluster. the no…

### PR DESCRIPTION
**Pull request by chia7712**
…deNames is what user input. And deadNodes is the (nodeNames - nodes hosting running container)

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2243)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

File: ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala [hunk 1]
```diff
@@ -179,16 +179,19 @@ trait StreamCollie extends Collie[StreamClusterInfo] {
     // get the first running container, or first non-running container if not found
     val first = containers.find(_.state == ContainerState.RUNNING.name).getOrElse(containers.head)
     val settings = seekSettings(first.environments)
+    // reuse the parser from Creation
+    val nodeNames = StreamApi.Creation(settings).nodeNames
+    println(s"[CHIA] containers:${containers.map(c => s"${c.nodeName} ${c.name} ${c.state}").mkString(",")}")
```
Comment (saivirtue) [inline ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala]: :)

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
retry it

**Comment by chia7712** [comment]
the flaky TestK8SStreamApp is traced by https://github.com/oharastream/ohara/issues/2393

**Pull request merged**